### PR TITLE
spanner: fix broken system test

### LINKED ISSUE #2397: System test failures on spanner
There are 4 failures in spanner

```
  1) Spanner Transactions read only should accept a read timestamp:

      Uncaught AssertionError: 5 === 1
      + expected - actual

      -5
      +1
      
      at system-test/spanner.js:2719:20
      at DestroyableTransform.<anonymous> (src/transaction.js:532:7)
      at endReadableNT (node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:920:12)

  2) Spanner Transactions read only should accept an exact staleness:

      Uncaught AssertionError: 5 === 2
      + expected - actual

      -5
      +2
      
      at system-test/spanner.js:2758:20
      at DestroyableTransform.<anonymous> (src/transaction.js:532:7)
      at endReadableNT (node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:920:12)

  3) Spanner Transactions read only should do an exact read with concurrent updates:

      Uncaught AssertionError: 'v4' === 'v3'
      + expected - actual

      -v4
      +v3
      
      at system-test/spanner.js:2850:24
      at DestroyableTransform.<anonymous> (src/transaction.js:532:7)
      at endReadableNT (node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:920:12)

  4) Spanner Transactions read only should read with staleness & concurrent updates:

      Uncaught AssertionError: 5 === 1
      + expected - actual

      -5
      +1
      
      at system-test/spanner.js:2878:22
      at src/session-pool.js:314:16
```

### PR DESCRIPTION
**Pull request by callmehiphop**
Closes #2397

### CODE REVIEW

File: packages/spanner/system-test/spanner.js [hunk 1]
```diff
@@ -2716,7 +2716,7 @@ var spanner = new Spanner(env);
           transaction.run('SELECT * FROM TxnTable', function(err, rows) {
             assert.ifError(err);
 
-            assert.strictEqual(rows.length, 2);
+            assert.strictEqual(rows.length, 1);
```
Comment (callmehiphop) [inline packages/spanner/system-test/spanner.js]: @stephenplusplus this was my fault. So when we capture the timestamp from inserts, they're approximate, in that we capture them before an insert happens, meaning when we grab the timestamp from insert `n`, it should return `n - 1` values.

Comment (callmehiphop) [inline packages/spanner/system-test/spanner.js]: However I am a little confused where 3 comes from in the failures you are seeing, locally I consistently get 1 returned value.

Comment (stephenplusplus) [inline packages/spanner/system-test/spanner.js]: I'll give it a run and see what we get.

Comment (stephenplusplus) [inline packages/spanner/system-test/spanner.js]: Still 3 consistently. Here are the rows being returned (`console.log(rows)` added on the line above this change):

```
[ [ { name: 'Key', value: 'k0' },
    { name: 'StringValue', value: 'v0' },
    { name: 'NumberValue', value: null } ],
  [ { name: 'Key', value: 'k1' },
    { name: 'StringValue', value: 'v1' },
    { name: 'NumberValue', value: null } ],
  [ { name: 'Key', value: 'k2' },
    { name: 'StringValue', value: 'v2' },
    { name: 'NumberValue', value: null } ] ]
```

Comment (stephenplusplus) [inline packages/spanner/system-test/spanner.js]: Note as well that @vkedia was receiving `5` in his original report from #2397.

Comment (callmehiphop) [inline packages/spanner/system-test/spanner.js]: I'm kind of at a loss because mine consistently returns 1 row.

Comment (stephenplusplus) [inline packages/spanner/system-test/spanner.js]: Have you tried running all of the Spanner system tests a couple times, not just the one?

Comment (swcloud) [inline packages/spanner/system-test/spanner.js]: In my case, with the this fix, the test passed. I run the test three times.

Comment (callmehiphop) [inline packages/spanner/system-test/spanner.js]: Yeah, I've tried various combinations of running just the single test, just txn tests and running all tests and it passes every time.

Comment (stephenplusplus) [inline packages/spanner/system-test/spanner.js]: Alright. I'll just not run these :)

**Pull request merged**